utils/node_local_test.py

```diff
@@ -5340,10 +5350,64 @@ def get_cmd(cont_id):
     test_cmd.wf = wf
     test_cmd.check_daos_stderr = True
     test_cmd.check_post_stdout = False
+    # Set the ignore_busy flag so that memory leaks on shutdown are ignored in some cases.
+    test_cmd.ignore_busy = True
 
     return test_cmd.launch()
 
 
+# Hack for test_alloc_fail_copy_trunc.  Variable scoping means that the get_cmd() function can read
+# but not modify variables in enclosing scope so keeping a counter has to be global.  As python
+# expects global variables to be constants pylint then checks this is upper case so disable that
+# warning.
+aftc_idx = 0  # pylint: disable=invalid-name
+
+
+def test_alloc_fail_copy_trunc(server, conf, wf):
+    """Run container (filesystem) copy under fault injection.
+
+    Use filesystem copy to truncate a file.
+
+    Create an initial container to modify, pre-populate it with a number of files of known length
+    then have each iteration of the test truncate one file.
+    """
+    # The number of files to pre-create.  This just needs to be bigger than the iteration count
+    # however too many will consume extra resources.
+    files_needed = 4000
+
+    def get_cmd(_):
+        global aftc_idx
+        cmd = [join(conf['PREFIX'], 'bin', 'daos'), 'filesystem', 'copy', '--src', src_file.name,
+               '--dst', f'daos://{pool.id()}/aftc/new_dir/file.{aftc_idx}']
+        aftc_idx += 1
+        assert aftc_idx <= files_needed
+        return cmd
```

Comment:
This also works:
```suggestion
    def get_cmd(_):
        cmd = [join(conf['PREFIX'], 'bin', 'daos'), 'filesystem', 'copy', '--src', src_file.name,
               '--dst', f'daos://{pool.id()}/aftc/new_dir/file.{get_cmd.aftc_idx}']
        get_cmd.aftc_idx += 1
        assert get_cmd.aftc_idx <= files_needed
        return cmd
    get_cmd.aftc_idx = 0
```


Simplified example:
```
def fun():
    def helper():
        print(helper.val)
        helper.val += 1
    helper.val = 0

    for _ in range(5):
        helper()
fun()
```

Reply:
I'm not familiar with the helper.val notation but I got it to work using similar but couldn't resolve the pylint or runtime warnings.

Reply:
Will update this PR, thank you.